### PR TITLE
[4.1] RavenDB-11913 - Support creating a database from directory of incremental backups

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11913

### CODE REVIEW

File: src/Raven.Client/Documents/Smuggler/BackupUtils.cs [hunk 1]
```diff
@@ -1,11 +1,32 @@
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 
 namespace Raven.Client.Documents.Smuggler
 {
-    internal static class BackupUtils
+    public static class BackupUtils
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/BackupUtils.cs]: Can't we stay with internal?

File: src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs [hunk 2]
```diff
@@ -20,6 +20,9 @@ public class DatabaseSmuggler
         private readonly string _databaseName;
         private readonly RequestExecutor _requestExecutor;
 
+        private const string LegacyIncrementalBackupExtension = ".ravendb-incremental-dump";
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs]: can be removed?

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs [hunk 3]
```diff
@@ -15,9 +15,8 @@ namespace Raven.Server.Documents.PeriodicBackup.Restore
     public static class RestoreUtils
     {
         private static readonly Regex BackupFolderRegex = new Regex(@".ravendb-(.+)-([A-Za-z]+)-(.+)$", RegexOptions.Compiled);
-        private const string LegacyIncrementalBackupExtension = "ravendb-incremental-dump";
-        private const string LegacyFullBackupExtension = "ravendb-full-dump";
-        
+        private static readonly Regex FileNameRegex = new Regex(@"([0-9]{4}-[0-9]{2}-[0-9]{2}-[0-9]{2}-[0-9]{2})", RegexOptions.Compiled);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs:18]: rename to LegacyFileNameRegex?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs:18]: this applies to our current backups as well (but should be very rare)

2018-01-01-13-01-1.ravendb-incremental-backup
2018-01-01-13-01-2.ravendb-incremental-backup

Comment (ppekrol) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreUtils.cs:18]: ok

File: src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs]: done

File: src/Raven.Client/Documents/Smuggler/BackupUtils.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Smuggler/BackupUtils.cs]: no, need to access it from RestoreUtils (Raven.Server)

Comment (ppekrol) [reply src/Raven.Client/Documents/Smuggler/BackupUtils.cs]: Internals are visible to Raven.Server

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Smuggler/BackupUtils.cs]: done

**Pull request merged**